### ISSUE TITLE
Replace memcached with redis as object cache system

### ISSUE THREAD

**Issue by AaronHolbrook** [Pilothouse-App/pilothouse#49]
Just a possibility, but in using both, I've been finding Redis to be much more capable and easy to deal with.

http://stackoverflow.com/questions/10558465/memcached-vs-redis

http://www.infoworld.com/article/3063161/application-development/why-redis-beats-memcached-for-caching.html

**Comment by philipnewcomer** [comment]
I have definitely seen a shift toward Redis as the preferred object cache for WordPress over Memcached and would be 👍 for this. My only concern in moving Dockerbox to Redis would be that many of our production sites are still on Memcached, and there is value in having our local environments match production as closely as possible.

However, Redis does have distinct advantages, and I'd like to switch over, so long as Memcached was still an option for those local sites that needed it. This could be as simple as documenting how to switch out Redis with Memcached in the wiki. However, this would depend on infrastructure added in #46.

Will revisit this issue once #46 is in place.

**Comment by AaronHolbrook** [comment]
I think the object caching layer is pretty well abstracted, and dependent on the environment. I guess you may need to provide the different `object-cache` file, but other than that it should function the same in the codebase: `wp_cache_get/set`.

**Comment by philipnewcomer** [comment]
This is a bit of an edge case, but it's a real issue I've dealt with previously in client work that illustrates why you'd want to be using the actual caching backend locally that you are on production:

https://github.com/markjaquith/WP-TLC-Transients/issues/29
Memcached doesn't store entries with dates longer than 30 days. As a result, something that worked with Redis locally could be broken on production with Memcached. Better to catch it locally before it gets to production.

**Comment by AaronHolbrook** [comment]
Ah, interesting.

I would try to always treat cache as something that is extremely volatile, but I can see where you're coming from.

My vote is to have Redis as the default and allow memcached to be the one that's custom placed in

**Comment by philipnewcomer** [comment]
Added to the `develop` branch as this is a breaking change. Also updated the PHP images in https://github.com/Pilothouse-App/pilothouse/commit/ee984985ca2011de8aca14443dc97a7edccdf280.

I decided to switch to Redis without leaving Memcached in place, as I don't see a good reason to have both object cache backends in place at the same time, and anyone who absolutely needs to use Memcached over Redis can install it manually.

Instructions for upgrading existing local sites using Memcached are here: https://github.com/Pilothouse-App/pilothouse/wiki/Upgrading-from-Memcached-to-Redis
